app/components/add-cron-job.js

```diff
@@ -1,45 +1,48 @@
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
-import { task } from 'ember-concurrency';
+import { task, timeout } from 'ember-concurrency';
+import { mapBy } from '@ember/object/computed';
+import Branch from 'travis/models/branch';
+import config from 'travis/config/environment';
 
 export default Component.extend({
   store: service(),
 
   classNames: ['form--cron'],
 
+  intervals: ['Monthly', 'Weekly', 'Daily'],
+
+  options: ['Always run', 'Do not run if there has been a build in the last 24h'],
+
+  currentCronJobsBranches: mapBy('repository.cronJobs', 'branch.name'),
+
+  init() {
+    this.reset();
+    this._super(...arguments);
+  },
+
   reset() {
     this.setProperties({
       selectedBranch: null,
-      selectedInterval: null,
-      disable: null
+      selectedInterval: this.intervals.firstObject,
+      selectedOption: this.options[0]
     });
   },
 
-  save: task(function* () {
-    const store = this.get('store');
-    const repoId = this.get('branches.firstObject.repoId');
-    const branch = this.get('selectedBranch') || this.get('branches.firstObject');
-
-    const existingCrons = yield store.filter('cron', { repository_id: repoId }, (c) => {
-      c.get('branch.repoId') === repoId && c.get('branch.name') === branch.get('name');
-    });
+  search: task(function* (query) {
+    yield timeout(config.intervals.searchDebounceRate);
+    let branchNames = this.get('currentCronJobsBranches');
+    let branches = yield Branch.search(this.store, query, this.repository.id);
+    return branches.reject(branch => (branchNames.indexOf(branch.name) > -1));
```

Comment:
Could you please use `branchNames.includes(branch.name)` instead of `indexOf`.

Reply:
Nope, IE9 and EDGE 12 is not supported

Reply:
And about 4 - this is the way how it is handled now

Reply:
`includes` will be polifilled by Ember: https://api.emberjs.com/ember/3.10/classes/EmberArray/methods/includes?anchor=includes

About 4 - It doesn't mean that currently it's the best solution. We should always try to improve our UX. And in my opinion, it fees confusing currently, when you know that the branch exists, but you can't find it in the list. For example, the most popular case would be with `master` branch. Please try to go [here](https://crons.test-deployments.travis-ci.org/travis-ci/travis-web/settings) and add a cron job for `master` and you'll see what I mean. And then try to do the same [here](https://as-crons-improvements.test-deployments.travis-ci.org/travis-ci/travis-web/settings) - doesn't it feel less confusing?
My concern is that this confusion will cause support issues.

Reply:
@weban please change to `includes` and let me know what you think about (4)